### PR TITLE
Move secrets

### PR DESCRIPTION
**Pull request by EBEN4REAL**
# Description 📣
This feature enables users perform bulk actions on secrets such as deleting multiple secrets and moving secrets into a different folder

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

image

[Feature Demo](https://share.vidyard.com/watch/WS18ETEHJykhsCVMxo7DDD?)

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/code-of-conduct). 📝

### CODE REVIEW

**Comment by maidul98** [comment]
Looks cool! I have requested @akhilmhdh  to look into this since he works closely with the frontend

**Comment by EBEN4REAL** [comment]
Alright

File: frontend/src/views/SecretMainPage/components/ActionBar/ActionBar.tsx [hunk 1]
```diff
@@ -186,9 +196,88 @@ export const ActionBar = ({
...
+
+  const checkedSecrets = Object.keys(selectedSecrets).map((secretId) => ({
+    _id: secretId,
+  }));
+
+
   return (
     <>
       <div className="flex items-center space-x-2 mt-4">
+        <div className={twMerge("fixed transform  flex justify-center opacity-0 bottom-[20px] left-[220px] scale-50 right-0 z-10 pointer-events-none translate-y-20  transition-all duration-300", Object.keys(selectedSecrets).length > 0 && "translate-y-0 scale-100 opacity-100")}>
```
Comment (akhilmhdh) [inline frontend/src/views/SecretMainPage/components/ActionBar/ActionBar.tsx:232]: multiple comments over this one

1. Missing permission check 
2. Avoid defining custom tailwind classes which already exist 
3. Move back to old multi select style please

File: frontend/src/views/SecretMainPage/components/ActionBar/ActionBar.tsx [hunk 2]
```diff
@@ -186,9 +196,88 @@ export const ActionBar = ({
...
+              <div className="bg-mineshaft-700 hover:bg-mineshaft-500 cursor-pointer flex justify-center items-center border border-mineshaft-500 rounded-md px-[15px] py-1.5 text-gray-200"
+                role="button"
+                onClick={() => handleMoveSecretsModalOpen()}
+                tabIndex={-1}
+                onKeyUp={() => { }}>
+                <FontAwesomeIcon icon={faUpDownLeftRight} className="mr-2.5" />
+                Move
+              </div>
+            </div>
+            <div role="button" className="bg-mineshaft-700 hover:bg-mineshaft-500  cursor-pointer flex justify-center items-center border rounded-md border-mineshaft-500 px-[15px] py-1.5 text-gray-200"
```
Comment (akhilmhdh) [inline frontend/src/views/SecretMainPage/components/ActionBar/ActionBar.tsx:254]: We have button components please misuse html ones

**Review by akhilmhdh — CHANGES_REQUESTED** [review]
Need to refactor to remove usage of folder id, bring back permission check and also please lint the code especially front-end

File: backend/src/controllers/v3/secretsController.ts [hunk 3]
```diff
@@ -839,3 +839,36 @@ export const deleteSecretByNameBatch = async (req: Request, res: Response) => {
...
   });
 };
+
+export const moveSecretsToFolder = async (req: Request, res: Response) => {
+  const {
+    body: { secrets, workspaceId, environment, folderId, secretPath }
+  } = await validateRequest(reqValidator.MoveSecretsToFolderV3, req);
+
+   if (req.user?._id) {
+    const { permission } = await getUserProjectPermissions(req.user._id, workspaceId);
```
[the remark this answers is not in the document]
Comment (maidul98) [reply backend/src/controllers/v3/secretsController.ts:849]: yes we need to check if the user has access to write `from` and `to`

File: backend/src/controllers/v3/secretsController.ts [hunk 4]
```diff
@@ -839,3 +839,36 @@ export const deleteSecretByNameBatch = async (req: Request, res: Response) => {
...
+    ForbiddenError.from(permission).throwUnlessCan(
+      ProjectPermissionActions.Edit,
+      subject(ProjectPermissionSub.Secrets, { environment, secretPath })
+    );
+  } else {
+    await validateServiceTokenDataClientForWorkspace({
+      serviceTokenData: req.authData.authPayload as IServiceTokenData,
```
[the remark this answers is not in the document]
Comment (maidul98) [reply backend/src/controllers/v3/secretsController.ts:856]: Agree, moving secrets should go through approvals if enabled for the environment

**Comment by maidul98** [comment]
The backend has moved to Postgres. This PR is now stale.

**Pull request closed without merging**